process/sync/baseSync.go

```diff
@@ -154,67 +157,49 @@ func (boot *baseBootstrap) processReceivedHeader(headerHandler data.HeaderHandle
 		log.Debug("forkDetector.AddHeader", "error", err.Error())
 	}
 
-	boot.mutRcvHdrHash.Lock()
-	hash := boot.requestedHeaderHash()
-	if hash == nil {
-		boot.mutRcvHdrHash.Unlock()
-		return
-	}
+	go boot.requestMiniBlocks(headerHandler.GetShardID(), headerHandler.GetNonce())
```

Comment:
why do we call requestMiniBlocks (we might have the miniblocks already in cache)?

Reply:
no because first we receive a header and after that we request miniBlocks

Reply:
The requests will be done only if they are not in pool and they are not already requested (controlled by RequestItemHandler implementation). We can receive here also headers requested in advance and we need to request then miniblocks and txs for each of them. They could not be in pool.

Reply:
But if they are (miniblocks/transactions got here before the header) we end up making useless network traffic